networkx/algorithms/tests/test_swap.py

```diff
@@ -12,13 +12,29 @@ def test_directed_edge_swap():
     assert out_degrees == sorted((n, d) for n, d in G.out_degree())
 
 
+def test_edge_cases_directed_edge_swap():
+    graph = nx.path_graph(4, create_using=nx.DiGraph)
+    with pytest.raises(nx.NetworkXAlgorithmError):
+        nx.directed_edge_swap(graph, nswap=4, max_tries=10, seed=1)
+    graph = nx.DiGraph()
+    edges = [(0, 0), (0, 1), (1, 0), (2, 3), (3, 2)]
+    graph.add_edges_from(edges)
```

Comment:
Just a nit, but we can save a few lines by creating the graph directly from the edgelist:

```suggestion
    graph = nx.DiGraph([(0, 0), (0, 1), (1, 0), (2, 3), (3, 2)])
```